src/BenchmarksApps/SslStream/Server/Program.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Net;
+using System.Net.Security;
+using System.Security.Authentication;
+using System.Net.Sockets;
+using System.CommandLine;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Crank.EventSources;
+
+using SslStreamServer;
+
+internal class Program
+{
+    private static async Task<int> Main(string[] args)
+    {
+        var rootCommand = new RootCommand("SslStream benchmark server");
+        OptionsBinder.AddOptions(rootCommand);
+        rootCommand.SetHandler<ServerOptions>(Run, new OptionsBinder());
+        return await rootCommand.InvokeAsync(args).ConfigureAwait(false);
+    }
+
+    static async Task Run(ServerOptions options)
+    {
+        var sslOptions = CreateSslServerAuthenticationOptions(options);
+        using var sock = new Socket(SocketType.Stream, ProtocolType.Tcp);
+        sock.Bind(new IPEndPoint(IPAddress.IPv6Any, options.Port));
+        sock.Listen();
+
+        Log($"Listening on {sock.LocalEndPoint}.");
+
+        CancellationTokenSource ctrlC = new CancellationTokenSource();
+        Console.CancelKeyPress += (s, e) =>
+        {
+            Log("Shutting down...");
+            e.Cancel = true;
+            ctrlC.Cancel();
+        };
+
+        try
+        {
+            while (!ctrlC.IsCancellationRequested)
+            {
+                var clientSock = await sock.AcceptAsync(ctrlC.Token).ConfigureAwait(false);
+                _ = Task.Run(() => ProcessClient(clientSock, options, sslOptions, ctrlC.Token));
+            }
+        }
+        catch (OperationCanceledException)
+        {
+            // Expected, just return
+        }
+
+        Log("Exitting...");
+    }
+
+    static async Task ReadWriteScenario(SslStream stream, ServerOptions options, CancellationToken cancellationToken)
+    {
+        static async Task WritingTask(SslStream stream, int bufferSize, CancellationToken cancellationToken)
+        {
+            if (bufferSize == 0)
+            {
+                return;
+            }
+
+            var sendBuffer = new byte[bufferSize];
+
+            try
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    await stream.WriteAsync(sendBuffer, cancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                // Expected, just return
+            }
+        }
+
+        byte[] recvBuffer = new byte[options.ReceiveBufferSize];
+
+        CancellationTokenSource cts = new CancellationTokenSource();
+        using var reg = cancellationToken.Register(() => cts.Cancel());
+
+        var writeTask = Task.Run(() => WritingTask(stream, options.SendBufferSize, cts.Token));
+
+        var recvBufer = new byte[options.ReceiveBufferSize];
```

Comment:
should we start reading at all, if `options.ReceiveBufferSize` is 0? Maybe we need to do the same check as with the writing task, and then wait on both of the tasks?